#### cd /stuff/Dropbox/DNA/designs/W40

Matan 2017 08 04

Based on Corinna's code

The following merely adds W40 from Goulong to the sequences in the dictionary

GAG TTC ACA AGA GTT CAC AA


This will attached on the horizontal SE of the origami belt.


In [1]:
from time import time
import random
import DNA as DNATemp
import DNAstruct as DNA
import numpy as np
import pandas as pd
from datetime import date

##  Load Exisiting SE Dictionary

In [3]:
dictionaryFilename = 'DNAdic.csv'
pdDict = pd.read_csv(dictionaryFilename,usecols=['Label','Sequence','SE-Length'])#,index=False)
sequences = pdDict.Sequence.values
#DNATemp.TMelt(pdDict[pdDict.Label=='P11'].Sequence.values[0])
pdDict

Label              Sequence  SE-Length
0      A8              GTAGATTG          8
1     Ap8              CAATCTAC          8
2      B8              GTTATGAG          8
3     Bp8              CTCATAAC          8
4      C8              GTATTGAG          8
5     Cp8              CTCAATAC          8
6      D8              ACCATCAT          8
7    DE18    ACCATCATCGCCGTGTAA         18
8   DEp18    TTACACGGCGATGATGGT         18
9     Dp8              ATGATGGT          8
10    F11           GATGGATTAGG         11
11    G11           CCTAATCCATC         11
12    J11           GTTGTTCATCT         11
13   Jp11           AGATGAACAAC         11
14    K11           CATCACTCTAA         11
15   Kp11           TTAGAGTGATG         11
16    L20  GAGTTCACAAGAGTTCACAA         20
17   Lp20  TTGTGAACTCTTGTGAACTC         20
18    P11           GTATTCGAGTT         11
19    Q11           AACTCGAATAC         11

## Add chosen seqeunces to dictionary file (see below)

In [4]:
L20 = 'GAGTTCACAAGAGTTCACAA'
Lp20 = DNA.complement(L20,reverse=True)

pdDictNew = pdDict.append([{'Label':'L20','Sequence':L20,'SE-Length':len(L20)},\
                           {'Label':'Lp20','Sequence':Lp20,'SE-Length':len(Lp20)}],\
                           ignore_index=True)



dictionaryFileNameNew = 'SEdictionary' +date.today().strftime("%Y-%m-%d")+'.csv'

pdDictNew = pdDictNew.sort('Label')
pdDictNew.to_csv(dictionaryFileNameNew)

pdDictNew

/home/matan/anaconda2/lib/python2.7/site-packages/ipykernel/__main__.py:10: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)


Label              Sequence  SE-Length
0      A8              GTAGATTG          8
1     Ap8              CAATCTAC          8
2      B8              GTTATGAG          8
3     Bp8              CTCATAAC          8
4      C8              GTATTGAG          8
5     Cp8              CTCAATAC          8
6      D8              ACCATCAT          8
7    DE18    ACCATCATCGCCGTGTAA         18
8   DEp18    TTACACGGCGATGATGGT         18
9     Dp8              ATGATGGT          8
10    F11           GATGGATTAGG         11
11    G11           CCTAATCCATC         11
12    J11           GTTGTTCATCT         11
13   Jp11           AGATGAACAAC         11
14    K11           CATCACTCTAA         11
15   Kp11           TTAGAGTGATG         11
16    L20  GAGTTCACAAGAGTTCACAA         20
20    L20  GAGTTCACAAGAGTTCACAA         20
17   Lp20  TTGTGAACTCTTGTGAACTC         20
21   Lp20  TTGTGAACTCTTGTGAACTC         20
18    P11           GTATTCGAGTT         11
19    Q11           AACTCGAATAC         11

## Define Parameters for New Sequences

In [5]:
# This cell generate a lot of 8b sequences.
#Eventually Two such strands will make a single strand displacment sequence 8b&8b
#make a list with complements.
NedWord = 3
stickyLength = 40
#toeHoldLength = 10
cgSticky = (.3,.4)
cgToeHold = (.5,.7)
             

#keep only those in temperature range of 20-23
Tmin = 35
Tmax = 40


##  Start generating many sequences and gradually exclude them by predefined constrains

In [ ]:
Nseq=10000
DNAlistSticky = [DNA.rand_seq(stickyLength,cgSticky) for i in range(Nseq)]


random.shuffle(DNAlistSticky)

DNAlist = DNAlistSticky
n=10
print  'First ',n,' of ', len(DNAlist)
for i in range(10): print DNAlist[i]

In [ ]:
#Remove sequences with repeats (AAAs etc)
DNAlist=[seq for seq in DNAlist if not DNA.reptest(seq,3)]
N=10
print  'First ',N,' of ', len(DNAlist), 'non 3 repeats'
for i in range(10): print DNAlist[i]
    

In [ ]:
## Remove Palindromic Sequences
DNAlist=[seq for seq in DNAlist if DNA.palin_check(seq,4)<0]
N=10
print  'First ',N,' of ', len(DNAlist), 'Non Palindromic'
for i in range(10): print DNAlist[i]

In [17]:
## Remove Sticky with more than 4 GCs (no tetras)
DNAlist=[seq for seq in DNAlist if DNA.noGCrep(seq[:stickyLength],4)]
N=10
print  'First ',N,' of ', len(DNAlist), 'Non GC Quadros'
for i in range(10): print DNAlist[i]

First  10  of  46 Non GC Quadros
CGCAATCGGTTCACCAGTTATTATCATTATTAAGTTCCAG
TGAGTCTACACTTAGTTACGCTATTCAGAACACAGATGAT
TGTCTAATAAGGCAATGTAAGGTGTAGCAAGTATAACCAC
TGTGATTAGGTGATTAGGAAGTCATATCTATTGTAGGCAC
GTACAATGCTTAGTTAGGTCTTATCTACGTGCAGAACATA
ATAAGAATAACATACTAGGTACACGCATCGGTGATAATAT
TAGACAAGGTAATATACTTCACAAGCGAACACCAATCATC
AAGAATGAGCACCACCAATTAACGTAAGTTGATAACACAC
ATGTAATGTTAGCAATACGAATGTTCTAGCTTCCAGCCAA
TTCACTTAATATAACGGTCAATCTCCATTCGAGTACGCTT


In [22]:

ledict = DNATemp.TMeltSelect(DNAlist, interv=(Tmin,Tmax))
print 'There are ', len(ledict), 'strands melting in the range', Tmin,'-',Tmax,'C'
for key in ledict:print key, ledict[key]

IOError: [Errno 2] No such file or directory: 'melttemp0.txt'

In [20]:
#Save only the strands that don't conflict with sequences (nor compliments)

cands =[]
for seq in DNAlist:
    dummy = True
    for cseq in sequences: 
        if not DNA.word_check(cseq,seq,NedWord): dummy=False;break;
    if dummy: cands.append(seq);
        
print  'First ',N,' of ', len(cands), 'candiates which are not complimantary to any of previous sequences (nor to m13):'
for i in range(min(len(cands),10)): print cands[i]

First  10  of  23 candiates which are not complimantary to any of previous sequences (nor to m13):
CGCAATCGGTTCACCAGTTATTATCATTATTAAGTTCCAG
TGTGATTAGGTGATTAGGAAGTCATATCTATTGTAGGCAC
ATAAGAATAACATACTAGGTACACGCATCGGTGATAATAT
AAGAATGAGCACCACCAATTAACGTAAGTTGATAACACAC
TCTCTTAGATAACTTGGATAGGAGCAACAGCTTAGAATAG
TATCTCCAATATGTTCTAGTCCGTAACTGTTCCTACAATC
TGGTAGCACATGGTACTATATCTTCTGACTCTAGCAATAA
CTGTTGACAAGTTATTAGTCTGTGTGTAAGAAGAACGCAT
TTCATATAATTCCAGAGTTCTATCAAGCTAATACAGATCG
TCTATACAATACAGAACTAATCACATACCATCTCGGTTAA


In [21]:
DNAlistTemp = ledict.keys()
cands=[DNAlistTemp[5], DNA.complement(DNAlistTemp[5])] #Make the list and its complements
dlist=DNAlistTemp[6:] #keeping the rest of the list for comparison
t0=time()
tLim=10
word=4

while len(cands)<30 and time()-t0<tLim:
    for i in range(len(dlist)):
        dnext=dlist[i]
        compl=DNA.complement(dnext, reverse=True)
        matchflag=False
        for seq in cands:
            if DNA.word_check(seq, dnext, word)>-1 and DNA.word_check(seq, compl, word)>-1:
                matchflag=True
                break
                
        if not matchflag:
            cands+=[dnext,compl]
            dlist=dlist[i+1:]
            print i
            break
print 'Time = ', time()-t0
print cands         


#print DNATemp.TMeltSelect(DNAlist[:30], interv=(20,21))

NameError: name 'ledict' is not defined

In [30]:
DNATemp.TMelt(pdDict[pdDict.Label=='F11'].Sequence.values[0])

{'H': -326040.0, 'S': -910.4, 'temp': 37.68}

In [24]:
for cand in cands: 
    print cand[:stickyLength], DNATemp.TMelt(cand[:stickyLength]), '<br>\n'
    #print cand[stickyLength:], DNATemp.TMelt(cand[stickyLength:]),'\n'

ACAGACATAGT {'H': -316844.0, 'S': -879.05, 'temp': 38.28} <br>

TGTCTGTATCA {'H': -318934.0, 'S': -885.32, 'temp': 38.4} <br>

AGATGAACAAC {'H': -328548.0, 'S': -915.84, 'temp': 38.45} <br>

GTTGTTCATCT {'H': -328548.0, 'S': -915.84, 'temp': 38.45} <br>

TTGAACTCCTA {'H': -314754.0, 'S': -873.2, 'temp': 38.03} <br>

TAGGAGTTCAA {'H': -314754.0, 'S': -873.2, 'temp': 38.03} <br>

TTCTAACGCTA {'H': -326876.0, 'S': -907.06, 'temp': 39.47} <br>

TAGCGTTAGAA {'H': -326876.0, 'S': -907.06, 'temp': 39.47} <br>

TACCATTGTCT {'H': -315590.0, 'S': -873.62, 'temp': 38.72} <br>

AGACAATGGTA {'H': -315590.0, 'S': -873.62, 'temp': 38.72} <br>

TATTCTGTCTC {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

GAGACAGAATA {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

AAGCAATAGTC {'H': -327294.0, 'S': -913.33, 'temp': 38.01} <br>

GACTATTGCTT {'H': -327294.0, 'S': -913.33, 'temp': 38.01} <br>

ATCTTCGGTAA {'H': -321024.0, 'S': -889.5, 'temp': 39.16} <br>

TTACCGAAGAT {'H': -321024.0, 'S': -889.5, 't

# Chosen sequences:

## J11 : GTTGTTCATCT {'H': -328548.0, 'S': -915.84, 'temp': 38.45} <br>

## K11 : CATCACTCTAA {'H': -323114.0, 'S': -906.22, 'temp': 36.15} <br>

## All Candidates


ACAGACATAGT {'H': -316844.0, 'S': -879.05, 'temp': 38.28} <br>

TGTCTGTATCA {'H': -318934.0, 'S': -885.32, 'temp': 38.4} <br>

AGATGAACAAC {'H': -328548.0, 'S': -915.84, 'temp': 38.45} <br>

GTTGTTCATCT {'H': -328548.0, 'S': -915.84, 'temp': 38.45} <br>

TTGAACTCCTA {'H': -314754.0, 'S': -873.2, 'temp': 38.03} <br>

TAGGAGTTCAA {'H': -314754.0, 'S': -873.2, 'temp': 38.03} <br>

TTCTAACGCTA {'H': -326876.0, 'S': -907.06, 'temp': 39.47} <br>

TAGCGTTAGAA {'H': -326876.0, 'S': -907.06, 'temp': 39.47} <br>

TACCATTGTCT {'H': -315590.0, 'S': -873.62, 'temp': 38.72} <br>

AGACAATGGTA {'H': -315590.0, 'S': -873.62, 'temp': 38.72} <br>

TATTCTGTCTC {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

GAGACAGAATA {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

AAGCAATAGTC {'H': -327294.0, 'S': -913.33, 'temp': 38.01} <br>

GACTATTGCTT {'H': -327294.0, 'S': -913.33, 'temp': 38.01} <br>

ATCTTCGGTAA {'H': -321024.0, 'S': -889.5, 'temp': 39.16} <br>

TTACCGAAGAT {'H': -321024.0, 'S': -889.5, 'temp': 39.16} <br>

TCTCATAGTTC {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

GAACTATGAGA {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

TATTCACCTCA {'H': -314754.0, 'S': -872.78, 'temp': 38.16} <br>

TGAGGTGAATA {'H': -314754.0, 'S': -872.78, 'temp': 38.16} <br>

TCGTCATCTTA {'H': -324368.0, 'S': -901.21, 'temp': 38.84} <br>

TAAGATGACGA {'H': -324368.0, 'S': -901.21, 'temp': 38.84} <br>

GTTGTTGTGAT {'H': -332310.0, 'S': -923.78, 'temp': 39.64} <br>

ATCACAACAAC {'H': -332310.0, 'S': -923.78, 'temp': 39.64} <br>

AGTAGTAGTCA {'H': -313918.0, 'S': -875.71, 'temp': 36.44} <br>

TGACTACTACT {'H': -313918.0, 'S': -875.71, 'temp': 36.44} <br>

TTACAACAGGA {'H': -318516.0, 'S': -881.14, 'temp': 39.27} <br>

TCCTGTTGTAA {'H': -318516.0, 'S': -881.14, 'temp': 39.27} <br>

CATCACTCTAA {'H': -323114.0, 'S': -906.22, 'temp': 36.15} <br>

TTAGAGTGATG {'H': -323114.0, 'S': -906.22, 'temp': 36.15} <br>
 

In [22]:
'''
# List existing Sticky Ends to verify mismatch!

#Previous Sequences not to be conflicted:
#The Sequence for the SLegs ticky End:

A = 'GTAGATTG' #As generated by SE8b
B = 'GTTATGAG' #As generated by SE8b
C = 'GTATTGAG' #As generated by SE8b
stemLegs  = 'GCTCTACTAT' #As generated by stem10b
stemArms =('AGGTATGA','CGTATTTA') #As generated by corinna back at the days.
stemSD = 'GGATAGTAGA'
CS = 'CGTAATGAAAGGCAGGGCTCTCTGGATTGACTGTGCGAAGGGTAGCGAT' #adopted from Lunta/Lang

with open('m13seq.txt','r') as f: m13=f.read().upper()

sequences = [A,B,C, stemArms[0],stemArms[1],stemLegs,stemSD,m13]
compseq = []
for seq in sequences: compseq.append(DNA.complement(seq, reverse=True));
for cseq in compseq: sequences.append(cseq)

'''

'''
#The dictionary

#The Sequence for the Sticky End:
A8 = 'GTAGATTG' #As generated by SE8b
B8 = 'GTTATGAG' #As generated by SE8b
C8 = 'GTATTGAG' #As generated by SE8b

D8 = 'ACCATCAT' #generaterd by strandDisplacement
E10 = 'CGCCGTGTAA' #generaterd by strandDisplacement

F11 = 'GATGGATTAGG' #Adpoted from Kun-Ta
P11 = 'GTATTCGAGTT' #Adpoted from Kun-Ta

G11 = DNA.complement(F11, reverse=True) 
Q11 = DNA.complement(P11, reverse=True) 

DE18 = D8+E10

Ap8 = DNA.complement(A8,reverse=True)
Bp8 = DNA.complement(B8,reverse=True)
Cp8 = DNA.complement(C8,reverse=True)

Dp8 = DNA.complement(D8,reverse=True)
DEp18 = DNA.complement(DE18,reverse=True)

stemLegs  = 'GCTCTACTAT' #As generated by stem10b
stemArms =('AGGTATGA','CGTATTTA') #As generated by corinna back at the days.
stemSD = 'GGATAGTAGA'
CS = 'CGTAATGAAAGGCAGGGCTCTCTGGATTGACTGTGCGAAGGGTAGCGAT' #adopted from Lunta/Lang

with open('m13seq.txt','r') as f: m13=f.read().upper()


stemsHands =("","")#('AGGTATGA','CGTATTTA') #Generated by stem8

bulge = 'T'*22 ##Add 22Ts between staple and to extend SE.(~5nm) 1nt length ~0.676nm. Persistence length 2.2nm R=(2.2*0.676*22)^0.5 

nicksShiftUp = 5
nicksShiftDown = 0



stemsLegs = stemsHands#('GGGGGGGGGG','GGGGGGG')#'GCTCTACTAT' generated by stem10

stemsMiddleHand = stemsHands#('GGGGGGGGGG','GGGGGGGGG') #stemHands


'''

'''#form and save SE dictionary

dictionary = {'A8':A8,\
                'Ap8':Ap8,\
                'B8':B8,\
                'Bp8':Bp8,\
                'C8':C8,\
                'Cp8':Cp8,\
                'D8':D8,\
                'Dp8':Dp8,\
                'DE18':DE18,\
                'DEp18':DEp18,\
                'F11':F11,\
                'G11':G11,\
                'P11':P11,\
                'Q11':Q11}

pdDict = pd.DataFrame({'Label':dictionary.keys(), 'Sequence':dictionary.values()})
#Add length column of each sequence
pdDict['SE-Length'] = [len(s) for s in pdDict['Sequence'].values]
dictionaryFileName = 'SEdictionary' +date.today().strftime("%Y-%m-%d")+'.csv'

pdDict = pdDict.sort('Label')
pdDict.to_csv(dictionaryFileName)
'''

'#form and save SE dictionary\n\ndictionary = {\'A8\':A8,                \'Ap8\':Ap8,                \'B8\':B8,                \'Bp8\':Bp8,                \'C8\':C8,                \'Cp8\':Cp8,                \'D8\':D8,                \'Dp8\':Dp8,                \'DE18\':DE18,                \'DEp18\':DEp18,                \'F11\':F11,                \'G11\':G11,                \'P11\':P11,                \'Q11\':Q11}\n\npdDict = pd.DataFrame({\'Label\':dictionary.keys(), \'Sequence\':dictionary.values()})\n#Add length column of each sequence\npdDict[\'SE-Length\'] = [len(s) for s in pdDict[\'Sequence\'].values]\ndictionaryFileName = \'SEdictionary\' +date.today().strftime("%Y-%m-%d")+\'.csv\'\n\npdDict = pdDict.sort(\'Label\')\npdDict.to_csv(dictionaryFileName)\n'

##The chosen:

D: ACCATCAT {'H': -214852.0, 'S': -595.65, 'temp': 22.59} <br>
E: CGCCGTGTAA {'H': -321860.0, 'S': -871.11, 'temp': 46.33} 

F: AAGCAAGA {'H': -222794.0, 'S': -616.55, 'temp': 24.97} <br>
G: GCGATTCAGC {'H': -325204.0, 'S': -887.41, 'temp': 44.5} 

H: TTCGTTCT {'H': -227392.0, 'S': -632.02, 'temp': 24.96} <br>
I: CGCTAAGTCG {'H': -326458.0, 'S': -896.61, 'temp': 42.88} 

##All Candidates:

ACAGACATAGT {'H': -316844.0, 'S': -879.05, 'temp': 38.28} <br>

TGTCTGTATCA {'H': -318934.0, 'S': -885.32, 'temp': 38.4} <br>

AGATGAACAAC {'H': -328548.0, 'S': -915.84, 'temp': 38.45} <br>

GTTGTTCATCT {'H': -328548.0, 'S': -915.84, 'temp': 38.45} <br>

TTGAACTCCTA {'H': -314754.0, 'S': -873.2, 'temp': 38.03} <br>

TAGGAGTTCAA {'H': -314754.0, 'S': -873.2, 'temp': 38.03} <br>

TTCTAACGCTA {'H': -326876.0, 'S': -907.06, 'temp': 39.47} <br>

TAGCGTTAGAA {'H': -326876.0, 'S': -907.06, 'temp': 39.47} <br>

TACCATTGTCT {'H': -315590.0, 'S': -873.62, 'temp': 38.72} <br>

AGACAATGGTA {'H': -315590.0, 'S': -873.62, 'temp': 38.72} <br>

TATTCTGTCTC {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

GAGACAGAATA {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

AAGCAATAGTC {'H': -327294.0, 'S': -913.33, 'temp': 38.01} <br>

GACTATTGCTT {'H': -327294.0, 'S': -913.33, 'temp': 38.01} <br>

ATCTTCGGTAA {'H': -321024.0, 'S': -889.5, 'temp': 39.16} <br>

TTACCGAAGAT {'H': -321024.0, 'S': -889.5, 'temp': 39.16} <br>

TCTCATAGTTC {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

GAACTATGAGA {'H': -321860.0, 'S': -904.13, 'temp': 35.57} <br>

TATTCACCTCA {'H': -314754.0, 'S': -872.78, 'temp': 38.16} <br>

TGAGGTGAATA {'H': -314754.0, 'S': -872.78, 'temp': 38.16} <br>

TCGTCATCTTA {'H': -324368.0, 'S': -901.21, 'temp': 38.84} <br>

TAAGATGACGA {'H': -324368.0, 'S': -901.21, 'temp': 38.84} <br>

GTTGTTGTGAT {'H': -332310.0, 'S': -923.78, 'temp': 39.64} <br>

ATCACAACAAC {'H': -332310.0, 'S': -923.78, 'temp': 39.64} <br>

AGTAGTAGTCA {'H': -313918.0, 'S': -875.71, 'temp': 36.44} <br>

TGACTACTACT {'H': -313918.0, 'S': -875.71, 'temp': 36.44} <br>

TTACAACAGGA {'H': -318516.0, 'S': -881.14, 'temp': 39.27} <br>

TCCTGTTGTAA {'H': -318516.0, 'S': -881.14, 'temp': 39.27} <br>

CATCACTCTAA {'H': -323114.0, 'S': -906.22, 'temp': 36.15} <br>

TTAGAGTGATG {'H': -323114.0, 'S': -906.22, 'temp': 36.15} <br>


In [35]:
import numpy as np
h = -230.E3 #[J/mol]
s=-734. #[J/molK]
c=1. #[1/u^3]
r = 8.
no=4.
nc=8.
nu = (1E-3)**3 #u^3
dt=h*r**2/s**3*(np.log(nu*c/4))**2*(1/no-1/nc)

tc = h/(s+r*np.log(nu*c/4)/nc)
to = h/(s+r*np.log(nu*c/4)/no)
ddtc=-r*dt*np.log(nu*c/2)/nc/s
ddto=-r*dt*np.log(nu*c/2)/no/s
print 'Colloid-Ori Melting Tc = ', round(tc-273,1), '\pm ', round(ddtc,2), 'C'
print 'Orig-Orig Melting To = ',  round(to-273,1), '\pm ', round(ddto,2), 'C'
print '\Delta T = ', round(dt,1)

Colloid-Ori Melting Tc =  31.2 \pm  -0.07 C
Orig-Orig Melting To =  22.5 \pm  -0.13 C
\Delta T =  2.3


In [33]:
import numpy as np
#GACACTTA {'H': -225720.0, 'S': -639.12, 'temp': 20.03}
h = -225.720E3 #[J/mol]
s=-639.12#[J/molK]
c=1. #[1/u^3]
r = 8.
no=4.
nc=8.
ncc = 40.

nu = (1E-3)**3 #u^3
dt=h*r**2/s**3*(np.log(nu*c/4))**2*(1/no-1/nc)

tc = h/(s+r*np.log(nu*c/4)/nc)
to = h/(s+r*np.log(nu*c/4)/no)
tcc = h/(s+r*np.log(nu*c/4)/ncc)

ddtc=-r*dt*np.log(nu*c/2)/nc/s
ddto=-r*dt*np.log(nu*c/2)/no/s
ddtcc = -r*dt*np.log(nu*c/2)/ncc/s

print 'Colloid-Ori Melting Tc = ', round(tc-273,1), '\pm ', round(ddtc,2), 'C'
print 'Orig-Orig Melting To = ',  round(to-273,1), '\pm ', round(ddto,2), 'C'
print 'Colloids-Colloids Melting To = ',  round(tcc-273,1), '\pm ', round(ddtcc,2), 'C'
print '\Delta T = ', round(dt,1)

Colloid-Ori Melting Tc =  68.4 \pm  -0.11 C
Orig-Orig Melting To =  57.3 \pm  -0.23 C
Colloids-Colloids Melting To =  77.7 \pm  -0.02 C
\Delta T =  3.4
